client/sftp.js

```diff
@@ -0,0 +1,83 @@
+import sftp from 'ssh2-sftp-client';
+
+export class SFTPClient {
+  constructor(timeout, logger, encoding) {
+    this.encoding = encoding;
+    this.timeout = timeout;
+    this.log = logger;
+
+    this.readOptions = {
+      readStreamOptions: {
+        flags: 'r',
+        encoding: this.encoding,
+      },
+      pipeOptions: {
+        end: false
+      }
+    };
+
+    this.client = new sftp();
+
+    this.connectionClosed = true;
+
+    this.client.on('close', this.closedConnection);
+    this.client.on('end', this.closedConnection);
+  }
+
+  async size(path) {
+    if (this.connectionClosed) throw Exception('Connect first in order to use this function');
+
+    this.log(`Getting size for file ${path}`);
+
+    return this.connection
+      .then(() => {
+        return this.client.stat(path);
+      })
+      .then((data) => {
+        this.log(`File size ${data.size}`);
+        return data.size;
+      })
+      .catch((err) => {
+        console.error(err.message);
+      });
+  }
+
+  async access(options = {}) {
```

Comment:
Maybe we should rename this method in both ftp clients to `connect` - I think `basic-ftp` client's use of `access` instead of `connect` is confusing.

Reply:
Will do, I also thought it was confusing, but left it because it was there already.